subsys/net/ip/l2/openthread/openthread.c

```diff
@@ -0,0 +1,348 @@
+		add_ipv6_addr_to_ot(ot_context);
+	} else if (mgmt_event == NET_EVENT_IPV6_MADDR_ADD) {
+		add_ipv6_maddr_to_ot(ot_context);
+	}
+}
+
+void otPlatRadioGetIeeeEui64(otInstance *instance, uint8_t *ieee_eui64)
+{
+	ARG_UNUSED(instance);
+
+	memcpy(ieee_eui64, ll_addr->addr, ll_addr->len);
+}
+
+void otTaskletsSignalPending(otInstance *instance)
+{
+	k_sem_give(&ot_sem);
+}
+
+void PlatformEventSignalPending(void)
+{
+	k_sem_give(&ot_sem);
+}
+
+void ot_state_changed_handler(u32_t flags, void *context)
+{
+	struct openthread_context *ot_context = context;
+
+	SYS_LOG_INF("State changed! Flags: 0x%08x Current role: %d",
+		flags, otThreadGetDeviceRole(ot_context->instance));
+
+	if (flags & OT_CHANGED_IP6_ADDRESS_REMOVED) {
+		SYS_LOG_DBG("Ipv6 address removed");
+		rm_ipv6_addr_from_zephyr(ot_context);
+		rm_ipv6_maddr_from_zephyr(ot_context);
+	}
+
+	if (flags & OT_CHANGED_IP6_ADDRESS_ADDED) {
+		SYS_LOG_DBG("Ipv6 address added");
+		add_ipv6_prefix_to_zephyr(ot_context);
+		add_ipv6_addr_to_zephyr(ot_context);
+		add_ipv6_maddr_to_zephyr(ot_context);
+	}
+
+}
+
+void ot_receive_handler(otMessage *aMessage, void *context)
+{
+	struct openthread_context *ot_context = context;
+
+	u16_t offset = 0;
+	u16_t read_len;
+	struct net_pkt *pkt;
+	struct net_buf *prev_buf = NULL;
+
+	pkt = net_pkt_get_reserve_rx(0, K_NO_WAIT);
+	if (!pkt) {
+		SYS_LOG_ERR("Failed to reserve net pkt");
+		otMessageFree(aMessage);
+		return;
+	}
+
+	while (1) {
+		struct net_buf *pkt_buf;
+
+		pkt_buf = net_pkt_get_frag(pkt, K_NO_WAIT);
+		if (!pkt_buf) {
+			SYS_LOG_ERR("Failed to get fragment buf");
+			net_pkt_unref(pkt);
+			otMessageFree(aMessage);
+			return;
+		}
+
+		read_len = otMessageRead(aMessage, offset,
+			pkt_buf->data, net_buf_tailroom(pkt_buf));
+
+		if (!read_len) {
+			net_buf_unref(pkt_buf);
+			break;
+		}
+
+		net_buf_add(pkt_buf, read_len);
+
+		if (!prev_buf) {
+			net_pkt_frag_insert(pkt, pkt_buf);
+		} else {
+			net_buf_frag_insert(prev_buf, pkt_buf);
+		}
+
+		prev_buf = pkt_buf;
+
+		offset += read_len;
+	}
+
+	otMessageFree(aMessage);
+
+	SYS_LOG_DBG("Injecting Ip6 packet to Zephyr net stack");
+
+#if defined(CONFIG_OPENTHREAD_L2_DEBUG_DUMP_IPV6)
+	dump_pkt("Received IPv6 packet", pkt);
+#endif
+
+	if (!pkt_list_is_full(ot_context)) {
+		if (net_recv_data(ot_context->iface, pkt) < 0) {
+			SYS_LOG_ERR("net_recv_data failed");
+			net_pkt_unref(pkt);
+			return;
+		}
+
+		pkt_list_add(ot_context, pkt);
+	} else {
+		SYS_LOG_INF("Pacet list is full");
+		net_pkt_unref(pkt);
+	}
+}
+
+static void openthread_process(void *context, void *arg2, void *arg3)
+{
+	struct openthread_context *ot_context = context;
+
+	while (1) {
+		while (otTaskletsArePending(ot_context->instance)) {
+			otTaskletsProcess(ot_context->instance);
+		}
+		PlatformProcessDrivers(ot_context->instance);
+
+		k_sem_take(&ot_sem, K_FOREVER);
+	}
+}
+
+static enum net_verdict openthread_recv(struct net_if *iface,
+	struct net_pkt *pkt)
```

Comment:
so as said above, parameter indentation below last '(' + 1 space is also applied to function declaration